### PR TITLE
Add first verbose tutorial

### PR DESCRIPTION
**Pull request by jjshoots**
# Description

This adds a tutorial inspired by CleanRL's Atari PPO code. It is very verbose and minimal exteral libraries are required.

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have run `pytest -v` and no errors are present.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I solved any possible warnings that `pytest -v` has generated that are related to my code to the best of my knowledge.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Make sure tutorial actually solves
- [ ] Add in rendering code

### CODE REVIEW

**Comment by WillDudley** [comment]
I think it would be good to have the following docstring at the start of the file.
```
"""This is a minimal example to show how to use PettingZoo with a reinforcement learning algorithm (PPO; implementation inspired by CleanRL)

Python version used: ???

Requirements:
pettingzoo == 1.21.0
???
"""
```

If you agree, please may you fill in the question marks?

**Comment by jjshoots** [comment]
Yeah I'm actually not done with it yet, I'll get it polished up more once it actually works.

File: tutorials/cleanrl.py [hunk 1]
```diff
@@ -1,3 +1,15 @@
+"""Basic code which shows what it's like to run PPO on the Pistonball env using the parallel API, this code is inspired by CleanRL.
+
+This code is exceedingly basic, with no logging or weights saving.
+The intention was for users to have a (relatively clean) ~200 line file to refer to when they want to design their own learning algorithm.
+
+Dependencies:
+- SuperSuit==3.6.0
+- numpy==1.23.2
+- torch==1.12.1
+- pettinzoo==1.22.0
```
Comment (WillDudley) [inline tutorials/cleanrl.py]: typo

File: tutorials/cleanrl.py [hunk 2]
```diff
@@ -1,3 +1,15 @@
+"""Basic code which shows what it's like to run PPO on the Pistonball env using the parallel API, this code is inspired by CleanRL.
+
+This code is exceedingly basic, with no logging or weights saving.
+The intention was for users to have a (relatively clean) ~200 line file to refer to when they want to design their own learning algorithm.
+
+Dependencies:
+- SuperSuit==3.6.0
+- numpy==1.23.2
```
Comment (WillDudley) [inline tutorials/cleanrl.py:8]: numpy being stated not needed. If concerned about numpy version control, move it to the last dependency. imo I'd remove it

**Comment by WillDudley** [comment]
Runs on my machine. Ubuntu thinks pygame isn't responding after ~30s of eval rendering for some reason, but that doesn't matter, it still runs.

My stdout for interest:
```
/home/will/PycharmProjects/PettingZoo/cleanrltut/bin/python /home/will/PycharmProjects/PettingZoo/tutorials/cleanrl.py 
Training episode 0
Episodic Return: 4.8469390869140625
Episode Length: 124

Value Loss: 521.2794189453125
Policy Loss: 28.20737648010254
Old Approx KL: 0.017104119062423706
Approx KL: 0.003589913249015808
Clip Fraction: 0.1032318376068376
Explained Variance: -0.0064754486083984375

-------------------------------------------

Training episode 1
Episodic Return: -15.214306831359863
Episode Length: 124

Value Loss: 94.8969955444336
Policy Loss: -12.974645614624023
Old Approx KL: -0.010465074330568314
Approx KL: 0.0027156174182891846
Clip Fraction: 0.07465277777777778
Explained Variance: -0.005328774452209473

-------------------------------------------


Process finished with exit code 0
```

**Comment by jjshoots** [comment]
@WillDudley Good to know, on my system it starts preparing for takeoff and freezes.

File: tutorials/cleanrl.py [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+            nn.MaxPool2d(2),
+            nn.ReLU(),
+            nn.Flatten(),
+            self.layer_init(nn.Linear(128 * 8 * 8, 512)),
+            nn.ReLU(),
+        )
+        self.actor = self.layer_init(nn.Linear(512, num_actions), std=0.01)
+        self.critic = self.layer_init(nn.Linear(512, 1))
+
+    def layer_init(self, layer, std=np.sqrt(2), bias_const=0.0):
```
[the remark this answers is not in the document]
Comment (jjshoots) [reply tutorials/cleanrl.py:43]: Yes I agree this should be done, will get to it soon ™.

**Comment by WillDudley** [comment]
Was keen to move forward with the docs, so this has been merged with https://github.com/Farama-Foundation/PettingZoo/pull/810 . Also did that tiny refactor

**Pull request closed without merging**